### PR TITLE
Expose event Observable

### PR DESCRIPTION
**Pull request by yuryybk**
Resolves https://github.com/mapbox/mapbox-gl-native-android-internal/issues/27

### CODE REVIEW

**Comment by yuryybk** [comment]
Still in progress to create [Value](https://github.com/mapbox/mapbox-gl-native-android-internal/blob/master/MapboxGLAndroidSDKCore/src/cpp/style/value.hpp) in `jvm` from `native` for ObservableEvent

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/ObservableEvent.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package com.mapbox.mapboxsdk.maps;
+
+import androidx.annotation.NonNull;
+
+import java.util.Date;
+
+/**
+ * @brief Generic Event type used to notify \sa Observer.
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/ObservableEvent.java:8]: We can drop the @brief and \sa

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  private static AtomicInteger idCounter = new AtomicInteger(0);
+  private final int id;
+
+  public Observer() {
+    id = idCounter.incrementAndGet();
+  }
+
+  /**
+   * @return observer id
+   * @brief Gets observer id
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java:21]: `@brief` and order: doc before `@return` section

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   */
+  public int getId() {
+    return id;
+  }
+
+  /**
+   * @param event an \sa Event
+   * @brief Notifies an Observer about an \sa Event.
+   */
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java:30]: `@brief` and order: doc before `@return` section

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java [hunk 4]
```diff
@@ -2136,6 +2136,42 @@ public boolean isUserAnimationInProgress() {
...
+  // Events Observer
+  //
+
+  /**
+   * Subscribes an Observer to a provided list of event types.
+   * Observable will hold a strong reference to an \sa Observer instance, therefore,
+   * in order to stop receiving notifications, caller must call unsubscribe with an
+   * \sa Observer instance used for an initial subscription.
+   *
+   * @param observer an \sa Observer
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java:2149]: remove \sa in javadoc

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/ObservableEvent.java (hunk 1 above)
Comment (yuryybk) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/ObservableEvent.java]: fixed

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java (hunk 3 above)
Comment (yuryybk) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java:32]: fixed

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java (hunk 4 above)
Comment (yuryybk) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: fixed

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java (hunk 2 above)
Comment (yuryybk) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Observer.java]: fixed

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java (hunk 4 above)
Comment (yuryybk) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: fixed

**Pull request merged**